tests/unit/test_extensionPoints.py

```diff
@@ -0,0 +1,255 @@
+#tests/unit/test_extensionPoints.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""Unit tests for the extensionPoints module.
+"""
+
+import unittest
+import extensionPoints
+
+class ExampleClass(object):
+	def method(self):
+		return 42
+
+def exampleFunc():
+	return 3.14
+
+class TestBoundMethodWeakref(unittest.TestCase):
+
+	def onDelete(self, weak):
+		self.deleted = weak
+
+	def setUp(self):
+		self.deleted = None
+		self.instance = ExampleClass()
+		self.weak = extensionPoints.BoundMethodWeakref(self.instance.method, self.onDelete)
+
+	def test_get(self):
+		"""Test that we get the right strong reference from the weak reference.
+		"""
+		method = self.weak()
+		self.assertEqual(method, self.instance.method)
+
+	def test_onDelete(self):
+		"""Test that the deletion callback gets called.
+		"""
+		del self.instance
+		self.assertEqual(self.deleted, self.weak)
+
+	def test_isWeak(self):
+		"""Test that this is actually a weak reference;
+		i.e. that it dies when the instance dies.
+		"""
+		del self.instance
+		method = self.weak()
+		self.assertIsNone(method)
+
+class TestCallWithSupportedKwargs(unittest.TestCase):
+
+	def test_supportsNoKwargs(self):
+		called = []
+		def h():
+			called.append(True)

```

Comment:
Technically, it's still captured by reference (as is everything in Python), but because the referenced object is immutable, it's conceptually captured by value. The same is true for floats, booleans, tuples and strings.

Not 100% sure what you mean by boxing, but you could do something like this if you wanted to be able to increment an int from an inner function:

```
def outer():
    class data:
        intVal = 0
    def inner():
        data.intVal += 1
    inner()
    return data.intVal
```

This would return `1`. There are more elegant, reusable ways of doing this too such as creating a `MutableInt` class which you instantiate and each instance has a different value. That's probably more what you mean by boxing.